monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -44,34 +44,58 @@ class HTTPIslandAPIClient(IIslandAPIClient):
     A client for the Island's HTTP API
     """
 
+    TOKEN_HEADER_KEY = "Authentication-Token"
+
     def __init__(
-        self, agent_event_serializer_registry: AgentEventSerializerRegistry, http_client: HTTPClient
+        self,
+        agent_event_serializer_registry: AgentEventSerializerRegistry,
+        http_client: HTTPClient,
+        otp: str,
     ):
         self._agent_event_serializer_registry = agent_event_serializer_registry
-        self.http_client = http_client
+        self._http_client = http_client
+        self._otp = otp
 
     def connect(
         self,
         island_server: SocketAddress,
     ):
-        self.http_client.connect(island_server)
+        try:
+            self._http_client.server_url = f"{island_server}/api/"
+            self._http_client.get("", params={"action": "is-up"})
+        except Exception as err:
+            logger.debug(f"Connection to {island_server} failed: {err}")
```

Comment:
NIT: Since we're reraising, and the exception type basically carries this information in it, I don't know that we need to log this here. Whomever is calling this can decide whether or not to log it.